syntax = "proto3";


message Record {
    message Owner {
        // Public key of the owner who owns the record
        string owner_id = 1;

        // Approximately when the owner was updated, as a Unix UTC timestamp
        uint64 timestamp = 2;
    }

    message History {
        // Coordinates are expected to be in millionths of a degree
        string reader_id = 1;
        string ant_id = 2;
        string situation = 3;
        int32 places = 4;
        int32 transitions = 5;
        repeated sint32 token = 6 [packed=true];
        repeated sint32 incidenceMatrix = 7 [packed=true];
        // Approximately when the location was updated, as a Unix UTC timestamp
        uint64 timestamp = 8;
    }

    // The user-defined natural key which identifies the object in the
    // real world (for example a serial number)
    string record_id = 1;
    string tag_id = 2;

    // Ordered oldest to newest by timestamp
    repeated Owner owners = 3;
    repeated History history = 4;
}


message RecordContainer {
    repeated Record entries = 1;
}
